src/App.tsx

```diff
@@ -1,56 +1,83 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
 import './App.scss';
+import cn from 'classnames';
 import { peopleFromServer } from './data/people';
+import { DropdownMenu } from './components/DropDownMenu';
+import { Person } from './types/Person';
+
+const debounce = <T extends (...args: never[]) => void>
+(fn: T, delay: number) => {
+  let timerID: number;
+
+  return (searchQuery: string) => {
+    clearTimeout(timerID);
+    timerID = setTimeout(fn, delay, searchQuery);
+  };
+};
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [search, setSearch] = useState('');
+  const [appliedSearch, setAppliedSearch] = useState('');
```

Comment:
I think there is some better names for this states. Such as query or searchQuery.